### ISSUE TITLE
Meryl failed with exit code 1, 'meryl histogram failed'

### ISSUE THREAD

**Issue by yukicyk** [marbl/canu#115]
Hi, 

I was trying to assemble some nanopore data with the following command but ended up with error, 

$ canu -p 835-pt -d 835-pt-2d genomeSize=2.6m -nanopore-raw 835_2D_poretools.fastq

What should I do? 

I have copied the error messages below, and a complete log in text.

cheers,
yuki
[log.txt](https://github.com/marbl/canu/files/234061/log.txt)

```
--Meryl attempt 0 begins.
----------------------------------------
-- Starting concurrent execution on Fri Apr 22 13:52:22 2016 with 783.3 GB free disk space (1 processes; 1 concurrently)

    /Users/OB/Desktop/Yuki/nanopore/data/pass/835-pt-2d/trimming/0-mercounts/meryl.sh 1 > /Users/OB/Desktop/Yuki/nanopore/data/pass/835-pt-2d/trimming/0-mercounts/meryl.000001.out 2>&1

-- Finished on Fri Apr 22 13:52:23 2016 (1 second) with 783.3 GB free disk space
----------------------------------------
-- Meryl attempt 1 begins.
----------------------------------------
-- Starting concurrent execution on Fri Apr 22 13:52:23 2016 with 783.3 GB free disk space (1 processes; 1 concurrently)

    /Users/OB/Desktop/Yuki/nanopore/data/pass/835-pt-2d/trimming/0-mercounts/meryl.sh 1 > /Users/OB/Desktop/Yuki/nanopore/data/pass/835-pt-2d/trimming/0-mercounts/meryl.000001.out 2>&1

-- Finished on Fri Apr 22 13:52:27 2016 (4 seconds) with 783.2 GB free disk space
----------------------------------------
--
-- meryl failed.
--
-- Meryl attempt 2 begins.
----------------------------------------
-- Starting concurrent execution on Fri Apr 22 13:52:27 2016 with 783.2 GB free disk space (1 processes; 1 concurrently)

    /Users/OB/Desktop/Yuki/nanopore/data/pass/835-pt-2d/trimming/0-mercounts/meryl.sh 1 > /Users/OB/Desktop/Yuki/nanopore/data/pass/835-pt-2d/trimming/0-mercounts/meryl.000001.out 2>&1

-- Finished on Fri Apr 22 13:52:28 2016 (1 second) with 783.3 GB free disk space
----------------------------------------
----------------------------------------
-- Starting command on Fri Apr 22 13:52:28 2016 with 783.3 GB free disk space

    /Users/OB/Desktop/Yuki/canu-1.2/Darwin-amd64/bin/meryl \
      -Dh \
      -s /Users/OB/Desktop/Yuki/nanopore/data/pass/835-pt-2d/trimming/0-mercounts/835-pt.ms22 \
    > /Users/OB/Desktop/Yuki/nanopore/data/pass/835-pt-2d/trimming/0-mercounts/835-pt.ms22.histogram \
    2> /Users/OB/Desktop/Yuki/nanopore/data/pass/835-pt-2d/trimming/0-mercounts/835-pt.ms22.histogram.info

-- Finished on Fri Apr 22 13:52:28 2016 (lickety-split) with 783.3 GB free disk space
----------------------------------------
ERROR:
ERROR:  Failed with exit code 1.
ERROR:
================================================================================
Please panic.  canu failed, and it shouldn't have.

Stack trace:

 at /Users/OB/Desktop/Yuki/canu-1.2/Darwin-amd64/bin/lib/canu/Defaults.pm line 234.
	canu::Defaults::caFailure('meryl histogram failed', '/Users/OB/Desktop/Yuki/nanopore/data/pass/835-pt-2d/trimming/...') called at /Users/OB/Desktop/Yuki/canu-1.2/Darwin-amd64/bin/lib/canu/Meryl.pm line 422
	canu::Meryl::merylProcess('/Users/OB/Desktop/Yuki/nanopore/data/pass/835-pt-2d', '835-pt', 'obt') called at /Users/OB/Desktop/Yuki/canu-1.2/Darwin-amd64/bin/canu line 483

Last few lines of the relevant log file (/Users/OB/Desktop/Yuki/nanopore/data/pass/835-pt-2d/trimming/0-mercounts/835-pt.ms22.histogram.info):

merylStreamReader()-- ERROR: /Users/OB/Desktop/Yuki/nanopore/data/pass/835-pt-2d/trimming/0-mercounts/835-pt.ms22.mcidx is not a merylStream index file!
merylStreamReader()-- ERROR: /Users/OB/Desktop/Yuki/nanopore/data/pass/835-pt-2d/trimming/0-mercounts/835-pt.ms22.mcdat is not a merylStream data file!

canu failed with 'meryl histogram failed'.
```

**Comment by skoren** [comment]
It looks like meryl is failing on your system. There should be a more detailed error in
```
/Users/OB/Desktop/Yuki/nanopore/data/pass/835-pt-2d/trimming/0-mercounts/meryl.000001.out
```

**Comment by brianwalenz** [comment]
The real error is being overwritten by a later (different) error.  You'll need to remove the *mcdat and *mcidx files (noted at the bottom of your report) and rerun:

/Users/OB/Desktop/Yuki/nanopore/data/pass/835-pt-2d/trimming/0-mercounts/meryl.sh 1

to generate the error.  Based on the report, it failed almost immediately.

**Comment by yukicyk** [comment]
Thank you for your comments.  

Here is a copy of the meryl.000001.out from this run
[meryl.000001.out.txt](https://github.com/marbl/canu/files/235816/meryl.000001.out.txt)

I actually solved the problem by running canu from bin/canu 
instead of calling it from a distant directory /Users/OB/Desktop/Yuki/canu-1.2/Darwin-amd64/bin/canu

Sorry for bothering if I have misused it.

**Comment by brianwalenz** [comment]
I'm not terribly satisfied.  It should have worked in either case.  If it fails again, and you can reproduce the failure, open a new issue and I'll add some debugging to track down the problem.

**Comment by yukicyk** [comment]
Hi Brian,

I went back to the failed folder and removed the *mcdat and *mcidx files, and reran meryl.sh 1, and no error generated this time. 

Threads all done, cleaning up.
Merge results.
      11.20 Mmers --  5.62 Mmers/second

I try canu again with the original setting and it succeeded without error, so the failure is not reproducible. I guess it was due to some random error in my system.

**Comment by cdb3ny** [comment]
Hello - I seem to be having a similar problem to the one above with canu1.2.  I've recently installed on several smaller servers with success in testing runs with ecoli example data.  However, now I'm getting setup to assemble a larger genome (1gb) on an IBM powerarch system (160 cores and 512gb ram).  I'm running all of these locally and not through a grid.  The test is also on on the ecoli dataset.  The correction step finishes appropriately.  Then it gets to the trimming and I get the same error above.  I have deleted the two files noted above, when I rerun the sh script - I get the message below.  It seems to have something to do with `_seqLen <= chunkLen', but the same assembly works find on other machines. Any thoughts would be greatly appreciated.
Donovan

[baileyd@ibm-power1 IBM]$ trimming/0-mercounts/./meryl.sh
Opened '/home/bpp/baileyd/Bailey_Lab_Space/programs/canu-1.2/selfSampleData-ecoli/IBM/trimming/ibm.gkpStore' with 13777 reads
Can fit 2319670864 mers into table with prefix of 25 bits, using 5734.000MB (0.000MB for positions)
Computing 4 segments using 4 threads and 316MB memory (294MB if in one batch).
  numMersActual      = 107064772
  mersPerBatch       = 26766193
  basesPerBatch      = 26838522
  numBuckets         = 2097152 (21 bits)
  bucketPointerWidth = 25
  merDataWidth       = 23
Computing segment 1 of 4.
 Allocating 73MB for mer storage (23 bits wide).
Computing segment 2 of 4.
 Allocating 73MB for mer storage (23 bits wide).
Computing segment 3 of 4.
 Allocating 73MB for mer storage (23 bits wide).
 Allocating 6MB for bucket pointer table (25 bits wide).
Computing segment 4 of 4.
 Allocating 73MB for mer storage (23 bits wide).
 Allocating 6MB for bucket pointer table (25 bits wide).
 Allocating 6MB for bucket pointer table (25 bits wide).
 Allocating 8MB for counting the size of each bucket.
 Allocating 6MB for bucket pointer table (25 bits wide).
 Allocating 8MB for counting the size of each bucket.
 Allocating 8MB for counting the size of each bucket.
 Allocating 8MB for counting the size of each bucket.
Opened '/home/bpp/baileyd/Bailey_Lab_Space/programs/canu-1.2/selfSampleData-ecoli/IBM/trimming/ibm.gkpStore' with 13777 reads
Opened '/home/bpp/baileyd/Bailey_Lab_Space/programs/canu-1.2/selfSampleData-ecoli/IBM/trimming/ibm.gkpStore' with 13777 reads
Opened '/home/bpp/baileyd/Bailey_Lab_Space/programs/canu-1.2/selfSampleData-ecoli/IBM/trimming/ibm.gkpStore' with 13777 reads
Opened '/home/bpp/baileyd/Bailey_Lab_Space/programs/canu-1.2/selfSampleData-ecoli/IBM/trimming/ibm.gkpStore' with 13777 reads
meryl: stores/gkStore.C:125: bool gkRead::gkRead_loadData(gkReadData*, void*): Assertion `_seqLen <= chunkLen' failed.
meryl: stores/gkStore.C:125: bool gkRead::gkRead_loadData(gkReadData*, void*): Assertion `_seqLen <= chunkLen' failed.
trimming/0-mercounts/./meryl.sh: line 36: 85928 Aborted                 (core dumped) $bin/meryl -B -C -L 2 -v -m 22 -threads 4 -memory 5734 -s /home/bpp/baileyd/Bailey_Lab_Space/programs/canu-1.2/selfSampleData-ecoli/IBM/trimming/ibm.gkpStore -o /home/bpp/baileyd/Bailey_Lab_Space/programs/canu-1.2/selfSampleData-ecoli/IBM/trimming/0-mercounts/ibm.ms22.WORKING
